File: app/src/main/java/com/walhalla/appextractor/activity/resources/p001/ManifestContract.java
package com.walhalla.appextractor.activity.resources.p001;

import android.content.Context;

import androidx.annotation.StringRes;
import androidx.fragment.app.FragmentActivity;

import com.walhalla.appextractor.activity.resources.ResItem;

import java.util.List;

public interface ManifestContract {
    interface View {
        void showResourceRawText(ResItem resource, String content);

        void showError(String message);

        void showSuccess(List<ResItem> list);

        void successToast(@StringRes int res, String aaa);

        void errorToast(@StringRes int res, String aaa);

    }

    interface Presenter {
        void loadManifestContent(String packageName);
        void saveAsset(Context context, ResItem resource);
        void exportIconRequest(Context context, ResItem resource);

        void readAssetRequest(Context context, ResItem resource);

        void zipAllAssetsRequest(FragmentActivity activity, ResItem resource);
    }
}

File: extractor/src/main/java/com/walhalla/appextractor/resources/ManifestContract.kt
package com.walhalla.appextractor.resources

import android.content.Context
import androidx.annotation.StringRes
import androidx.fragment.app.FragmentActivity

interface ManifestContract {

    interface View {
        fun showResourceRawText(resource: ResItem, content: String)

        fun showError(message: String)

        fun showSuccess(list: List<ResItem>)

        fun successToast(@StringRes res: Int, aaa: String)

        fun errorToast(@StringRes res: Int, aaa: String)
    }

    interface Presenter {
        fun loadManifestContent(packageName: String)
        fun saveAsset(context: Context, resource: ResItem)
        fun exportIconRequest(context: Context, resource: ResItem)

        fun readAssetRequest(context: Context, resource: ResItem)

        fun zipAllAssetsRequest(activity: FragmentActivity, resource: ResItem)
    }
}